crates/ra_hir/src/source_binder.rs

```diff
@@ -140,6 +140,11 @@ impl Expansion {
         exp_info.map_token_down(token)
     }
 
+    pub fn source(&self, db: &impl HirDatabase) -> Source<AstId<ast::MacroCall>> {
+        let loc = db.lookup_intern_macro(self.macro_call_id);
+        Source::new(self.file_id(), loc.ast_id)
+    }
```

Comment:
I don't know whether it is the correct way to get the source from `Expansion` ...

Reply:
We actually never use the value of `Source` here, so just making `file_id` public again (and using explicit `Source::new` at the call site) seems simpler